sdl-core/src/main/scala/io/smartdatalake/workflow/action/generic/transformer/EncryptColumnsTransformer.scala

```diff
@@ -106,19 +123,27 @@ trait EncryptDecrypt {
  * @param encryptColumns List of columns [columnA, columnB] to be encrypted
  * @param keyVariable    contains the id of the provider and the name of the secret with format <PROVIDERID>#<SECRETNAME>,
  *                       e.g. ENV#<ENV_VARIABLE_NAME> to get a secret from an environment variable OR CLEAR#mYsEcReTkeY
+ * @param algorithm      There are GCM (AES/GCM/NoPadding) and ECB (AES/ECB/PKCS5Padding) implemented. DEFAULT: GCM
  */
 case class EncryptColumnsTransformer(override val name: String = "encryptColumns",
                                      override val description: Option[String] = None,
                                      encryptColumns: Seq[String],
                                      @Deprecated @deprecated("Use `key` instead", "2.5.0") private val keyVariable: Option[String] = None,
                                      private val key: Option[StringOrSecret],
+                                     algorithm: String = "GCM"
                                     )
   extends SparkDfTransformer with EncryptDecrypt {
   private val cur_key: StringOrSecret = key.getOrElse(SecretsUtil.convertSecretVariableToStringOrSecret(keyVariable.get))
 
   override def keyAsBytes: Array[Byte] = cur_key.resolve().getBytes
 
-  override val cryptUDF: UserDefinedFunction = udf(encrypt _)
+  override val cryptUDF: UserDefinedFunction = algorithm match {
+    case "GCM" => udf(encryptGCM _)
+    case "ECB" => udf(encryptECB _)
+    case _ => {assert(true, s"unsupported en/decryption algorithm ${algorithm}, using GCM")
```

Comment:
So basically your coughs just add a warning instead of making an assert

Reply:
Here is probably more an exception than an assert. Assertion should not be used for production code and used instead of exceptions. Assertion can be disabled depending of the configuration to have performance benefits.

Reply:
thanks for the hints, indeed this needs to be an exception, I corrected it.